android/src/com/mapswithme/maps/MwmApplication.java

```diff
@@ -58,6 +58,8 @@
   private Logger mLogger;
   public final static String TAG = "MwmApplication";
 
+  private static MwmApplication instance;
```

Comment:
Don't add this static field please. Use the static method `from(Context context)` where you need the application context.

Reply:
The problem is that you need to check all places where `from(Context context)` is called to make sure that context is not null. It is inconvenient and error-prone. `MwmApplication` is a singleton by Android design and I see no problem with having this static field.